var/spack/repos/builtin/packages/faircmakemodules/package.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Faircmakemodules(CMakePackage):
+    """CMake Modules developed in the context of various FAIR (https://www.gsi.de/en/researchaccelerators/fair) projects"""
+
+    homepage = "https://fairrootgroup.github.io/FairCMakeModules/latest/"
+    url = "https://github.com/FairRootGroup/FairCMakeModules/archive/refs/tags/v1.0.0.tar.gz"
+    git = "https://github.com/FairRootGroup/FairCMakeModules.git"
+    maintainers("dennisklein", "ChristianTackeGSI")
+
+    version("develop", branch="main", get_full_repo=True)
```

Comment:
We prefer the version name match the branch name.  IIRC there was a reason for them to be different but that no longer applies.

Reply:
The most recent discussion on the explanation that I've seen is from @adamjstewart at https://github.com/spack/spack/pull/33920#discussion_r1023300181:

```
Once upon a time, Spack only considered develop to be newer than stable releases, and considered all other string versions to be older than all stable releases. This obviously caused problems, so we used to recommend people call these kinds of versions develop instead of master. Several years ago, we finally added master and several other [blessed strings](https://github.com/spack/spack/blob/v0.19.0/lib/spack/spack/version.py#L64) to the list of versions newer than release versions, so **now the correct policy is to keep the version name the same as the branch name unless it isn't in this list.
```

Reply:
Hmm.  I normally paste that link when I make suggestions like this.

The ordering for `when` and `satisfies` arguments still holds so, given you're only providing one branch, I'm not sure the issue with the requested change.  When I `spack install faircmakemocules`, Spack installs `1.0.0` whether this one branch is called `develop` or more accurately `main`.

Reply:
> Also the names did have some ordering, see Packaging Guide, scroll a little down to numbered point 1.
If that documentation is not correct any more, please highly consider updating the packaging guide.

Packaging Guide is still correct, both develop and main are newer than all stable releases, so if you name your version main it will sort correctly.

Reply:
Okay, understood. I renamed the version. Please re-review.